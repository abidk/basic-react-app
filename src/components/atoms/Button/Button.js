import React, { memo } from 'react';
import PropTypes from 'prop-types';
import './Button.css';

const Button = (props) => <button className="Button" onClick={props.action}>{props.label}</button>

Button.propTypes = {
    label: PropTypes.string.isRequired,
    action: PropTypes.func.isRequired
}

export default memo(Button);
