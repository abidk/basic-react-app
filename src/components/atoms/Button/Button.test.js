import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Button from './Button';

describe(__filename, () => {
  it('Should render', () => {
    const output = shallow(<Button label="Click here" action={() => { }} />);

    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('Should click event', () => {
    const mockCallBack = jest.fn();

    const button = shallow(<Button label="Click here" action={mockCallBack} />);
    button.find('button').simulate('click');

    expect(mockCallBack).toHaveBeenCalled();
  });
});