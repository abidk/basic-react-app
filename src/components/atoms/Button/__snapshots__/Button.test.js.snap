// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/abidkhalil/Documents/basic-react-app/src/components/atoms/Button/Button.test.js Should render 1`] = `
<button
  className="Button"
  onClick={[Function]}
>
  Click here
</button>
`;
